gnomad/utils/file_utils.py

```diff
@@ -100,23 +102,62 @@ def parallel_file_exists(fpaths: List[str], parallelism: int = 750) -> Dict[str,
     )
 
 
-def file_exists(fname: str) -> bool:
+def file_exists(fname: str, overwrite: Optional[bool] = None) -> bool:
     """
     Check whether a file exists.
 
     Supports either local or Google cloud (gs://) paths.
-    If the file is a Hail file (.ht, .mt extensions), it checks that _SUCCESS is present.
+    If the file is a Hail file (.ht, .mt, .bm, .parquet, and .vds extensions), it checks that _SUCCESS is present.
 
-    :param fname: File name
-    :return: Whether the file exists
+    :param fname: File name.
+    :param overwrite: Optional boolean that will raise an error if set to False and file exists.
+    :return: Whether the file exists.
     """
     fext = os.path.splitext(fname)[1]
-    if fext in [".ht", ".mt"]:
-        fname += "/_SUCCESS"
+    if fext in {".ht", ".mt", ".bm", ".parquet"}:
+        paths = [f"{fname}/_SUCCESS"]
+
+    if fext == ".vds":
+        paths = [f"{fname}/reference_data/_SUCCESS", f"{fname}/variant_data/_SUCCESS"]
+
     if fname.startswith("gs://"):
-        return hl.hadoop_exists(fname)
+        exists_func = hl.hadoop_exists
     else:
-        return os.path.isfile(fname)
+        exists_func = os.path.isfile
+
+    exists = all([exists_func(p) for p in paths])
+
+    if exists and overwrite is False:
+        raise DataException(
+            f"{fname} already exists and the overwrite option was not used!"
+        )
+
+    return exists
+
+
+def file_list_exists(
+    fname_list: List[str],
+    overwrite: bool = None,
+) -> bool:
+    """
+    Check whether all files in the list exist and raise an exception if any of them exist and overwrite is False.
+
```

Comment:
I feel that the `overwrite` parameter needs use case explanation, since overwrite is kind of out of the functions scope (just to check if file in list exists) and it causes some confusion. Maybe just me, so it's up to you :)
```suggestion
The function will optionally raise an error if `overwrite` is set to False and any file exists, when the the function is called to check if the new created file paths exist.
```